tests/dotnet/UnitTests/ProjectTest.cs

```
@@ -682,6 +684,173 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+
+		void DeleteAssets (string project_path)
+		{
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+			try {
+				if (Directory.Exists (xcassetsDir))
+					Directory.Delete (xcassetsDir, true);
+			}
+			catch (Exception e){
+				Console.WriteLine($"Deleting Assets.xcassets failed: {e.Message}");
+			}
+		}
+
+		void CopyAssets (string project_path)
+		{
+			DirectoryInfo testingAssetsDir = new DirectoryInfo (Path.Combine (project_path, @"../../TestingAssets"));
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+
+			if (!testingAssetsDir.Exists)
+				throw new DirectoryNotFoundException ($"TestingAssets directory does not exist or could not be found: {testingAssetsDir}");
+
+			if (!Directory.Exists (xcassetsDir))
+				CopyDirectoryContents (testingAssetsDir, xcassetsDir);
+		}
+
+		void CopyDirectoryContents (DirectoryInfo dir, string DestDirectory) {
+			Directory.CreateDirectory (DestDirectory);
+			foreach (FileInfo file in dir.GetFiles ()) {
+				string tempPath = Path.Combine (DestDirectory, file.Name);
+				file.CopyTo (tempPath, false);
+			}
+			foreach (DirectoryInfo subDir in dir.GetDirectories ()){
+				CopyDirectoryContents (subDir, Path.Combine (DestDirectory, subDir.Name));
+			}
+		}
+
+		JsonDocument ProcessAssets (string assetsPath) {
+			try {
+				using (Process myProcess = new Process ()) {
+					myProcess.StartInfo.UseShellExecute = false;
+					myProcess.StartInfo.RedirectStandardOutput = true;
+					myProcess.StartInfo.Arguments = $"--sdk iphoneos assetutil --info {assetsPath}";
+					myProcess.StartInfo.FileName = "xcrun";
+					myProcess.StartInfo.CreateNoWindow = true;
+					myProcess.Start ();
+
+					string output = myProcess.StandardOutput.ReadToEnd ();
+					StreamReader reader = myProcess.StandardOutput;
+					myProcess.WaitForExit ();
+
+					return JsonDocument.Parse (output);
+				}
+			}
+			catch (Exception e) {
+				Console.WriteLine($"Couldn't process: {e.Message}");
+			}
```

As Manuel said, don't swallow exceptions, it might hide test failures.